Update settings to latest version of gitlab api.

jabby opened the pull request:
According to the documentation there is many new settings available.
This commit update all settings and documentation to the latest version.

It's actually a work in progress as there is few TODOs in the code.

Basically the first verification to do is to validate that this modification is retro-compatible
with olders version of library.

I will update this PR next day. Feel free to comment my work in progress.

gmessner commented on src/main/java/org/gitlab4j/api/models/Setting.java:
```diff
@@ -67,54 +108,60 @@
-     * for the client side.
+     * Container Registry token duration in minutes.
      */
-    CLIENTSIDE_SENTRY_ENABLED(Boolean.class),
```
This should be deprecated not removed.

jabby replied:
I updated my patch. It was a mistake.

gmessner commented on src/main/java/org/gitlab4j/api/models/Setting.java:
```diff
@@ -67,54 +108,60 @@
-     * required by: clientside_sentry_enabled Clientside Sentry Data Source Name.
+     * Custom hostname (for private commit emails).
      */
-    CLIENTSIDE_SENTRY_DSN(String.class),
```
This should be deprecated not removed.

jabby replied:
I updated my patch. It was a mistake.

gmessner commented on src/main/java/org/gitlab4j/api/models/Setting.java:
```diff
@@ -217,485 +307,629 @@
+     * -1 disables RSA keys.
      */
-    RSA_KEY_RESTRICTION(Integer.class),
+    RSA_KEY_RESTRICTION(Boolean.class),
```
The documentation indicates that ```0``` or ```-1``` is returned, why change this to ```Boolean.class````

jabby replied:
Good catch. My bad.

jabby commented:
@gmessner I updated the change accordingly to your comments. I also updated few more things and dig a little for fields with no documentation. I updated javadoc when I found some useful informations. 

There is one task to do before merging this patch. See https://github.com/gitlab4j/gitlab4j-api/pull/462/files#diff-999bbdddeac9c1f965364d46f7c74cdeR88
I don't know which type we should use here.

gmessner commented:
@jabby 
I have a way to have it support both ```String``` and ```String[]```, but it would be more work for me to explain how to do it rather than do it myself.

My suggestion would be for you to finish everything else and let me know it is ready to be merged.  Once merged I will add support for the dual type and keep you updated on how it was accomplished.

jabby commented:
For my point of view, except about the support of `String` and `String[]`, this patch can be merged.
So feel free to merged and provide the additional change. I will check your modification and learn how you manage this case.

gmessner commented:
@jabby 
I am currently traveling, will be back late 11/6/2019 will merge then and add support for the ```String/String[]``` dual type.

jabby commented:
Ok no problem. Have a nice travel.
